File: lists/arrays/2Sum/solution.py
from typing import List


def twoSum(nums: List[int], target: int) -> List[int]:
    ht = {}

    for i in range(0, len(nums)):
        diff = target - nums[i]
        if diff in ht and ht[diff] != i:
            return [i, ht[diff]]
        ht[nums[i]] = i

    return []

File: lists/arrays/2Sum/solution.js
const assert = require('assert').strict;

const twoSum = function(nums, target) {
    const ht = new Map();
    const indexes = [];

    for (let i = 0; i < nums.length; i++) {
        const val = target - nums[i]
        if (ht.has(val)) {
            return [ht.get(val), i]
        }

        ht.set(nums[i], i)
    }
};

assert.deepEqual(twoSum([2, 7, 11, 15], 9), [0, 1]);
assert.deepEqual(twoSum([3, 2, 4], 6), [1, 2]);
assert.deepEqual(twoSum([3, 3], 6), [0, 1]);